File: Java/1121.java
__________________________________________________________________________________________________
class Solution {    
    public boolean canDivideIntoSubsequences(int[] nums, int K) {
        Map<Integer, Integer> cnt = new HashMap<>();
        int mx = 0;
        for (int i = 0; i < nums.length; i++) {
            cnt.put(nums[i], cnt.getOrDefault(nums[i], 0) + 1);
            mx = Math.max(mx, cnt.get(nums[i]));
        }
        
        if ((long) mx * K > nums.length) return false;
        return true;
    }
}
__________________________________________________________________________________________________

__________________________________________________________________________________________________

File: Python3/1121.py
__________________________________________________________________________________________________
class Solution:
    def canDivideIntoSubsequences(self, nums: List[int], K: int) -> bool:
        n=len(nums);tmp=1
        for i in range(1,n):
            if nums[i]!=nums[i-1]:
                if tmp*K>n:
                    return False
                tmp=1
            else:
                tmp+=1
        if tmp*K>n:
            return False
        return True
__________________________________________________________________________________________________

__________________________________________________________________________________________________
